scripts/rails_console/deduplicate_question_uids.rb

```diff
@@ -0,0 +1,397 @@
+ {:uid=>"-Jzw0qjP5tpJMplbm7lF", :count=>2338, :new_uid=>"17e7cc9d-e8cd-435e-b21b-140af1a58b52"},
+ {:uid=>"-Jzw0qjP5tpJMplbm7lG", :count=>2618, :new_uid=>"07c7882c-bb15-4abe-b467-ee0880a1ed01"},
+ {:uid=>"-Jzw0qjP5tpJMplbm7lH", :count=>564, :new_uid=>"e91aeb0a-4054-4913-bbb0-55a3a4d6a0e6"},
+ {:uid=>"-Jzw0qjP5tpJMplbm7lI", :count=>557, :new_uid=>"bad63a04-1ef6-4626-87fa-2d51affacdc7"},
+ {:uid=>"-Jzw0qjP5tpJMplbm7lJ", :count=>438, :new_uid=>"06ca16e4-0783-45c4-8cdd-ddd73ae04262"},
+ {:uid=>"-Jzw0qjP5tpJMplbm7lK", :count=>3329, :new_uid=>"c5d58592-82a8-4309-b69f-e2cd18b51519"},
+ {:uid=>"-Jzw0qjP5tpJMplbm7lL", :count=>3213, :new_uid=>"23102213-a9fc-436e-a900-b443c75c887d"},
+ {:uid=>"-Jzw0qjP5tpJMplbm7lX", :count=>146, :new_uid=>"e2d0cfcd-9574-48f8-a8bb-85cca323101f"},
+ {:uid=>"-Jzw0qjP5tpJMplbm7lZ", :count=>1, :new_uid=>"a2bc7f67-c399-4389-9ee0-451f4433c242"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmVW", :count=>8594, :new_uid=>"51ea24ad-44ab-4b60-b81c-ee3b67c5a83f"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmVZ", :count=>451, :new_uid=>"c0f74434-2121-4e81-baaf-27b33e1a0df4"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmWb", :count=>3185, :new_uid=>"415c2506-1690-47a8-b769-3890d913ff81"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmWD", :count=>4946, :new_uid=>"2ba0dbdc-8c26-4e09-9e8d-5a88a8418a59"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmWF", :count=>2071, :new_uid=>"4be830d6-dcaa-4fad-878e-f558fdcde6ab"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmWG", :count=>7050, :new_uid=>"50abf821-95cb-4af9-941e-b2db98946433"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmWH", :count=>2711, :new_uid=>"bc546b98-6e7e-48a9-8a61-172b1e16ac18"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmWK", :count=>14617, :new_uid=>"6045a79a-9d3f-4d9c-9d96-46ac5f4f4a00"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmWl", :count=>14084, :new_uid=>"9162ad57-9ff1-4fbb-b8b8-ff48bdf245bf"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmWn", :count=>9588, :new_uid=>"da695b69-66cb-4857-bf9b-6d352a03b046"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmWp", :count=>7021, :new_uid=>"7800ddac-a0d7-49f3-8d07-492b7442b323"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmWq", :count=>10621, :new_uid=>"e845292f-7304-4369-9253-5c95270ae482"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmWs", :count=>8693, :new_uid=>"807821c2-19c0-4272-ab50-6722c86a6f99"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmWu", :count=>10032, :new_uid=>"22591c57-d927-46c6-a28a-35b7e97e7a7b"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmWw", :count=>6131, :new_uid=>"9071a5f4-ca5c-4c7b-bc7a-20f187991f77"},
+ {:uid=>"-Jzw0qjQwbyM2twtkmWZ", :count=>3460, :new_uid=>"74448d32-1aa1-4ead-bb05-b2fa4d95207f"}]
+=end
+
+# took 13 minutes to run this in staging
+start_time = Time.now
+lower_count_questions.each do |question|
+  Response.where(question_uid: question[:uid]).find_in_batches(batch_size: 100) do |batch|
+    begin
+      responses_for_insertion = []
+      batch.each do |response|
+        duplicated_response = response.attributes.slice('author', 'text', 'feedback', 'count', 'first_attempt_count', 'child_count', 'optimal', 'weak', 'concept_results', 'spelling_error')
+
+        duplicated_response['question_uid'] = question[:new_uid]
+
```

Comment:
I think you want a temp variable for Time.now so that `created_at` and `updated_at` are exactly the same on creation:

```
  right_now = Time.now
  duplicated_response['created_at'] = right_now
  duplicated_response['updated_at'] =  right_now
```

Reply:
Additional item for consideration (I haven't fully thought this through, so take it with a grain of salt): should we set the `created_at` of the new response to the same value as the original response?  Just so that we maintain the understanding of when this text actually entered our system as a response?  I could see this going either way.  Even if we do copy the `created_at`, I do kind of like the idea of setting `updated_at` to `Time.now`.

Reply:
I think this is a good question -- I added it to the agenda for tomorrow's engineering meeting because I think it's worth the team agreeing on a general approach to similar issues.